WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift

```diff
@@ -0,0 +1,255 @@
+    /// There's no scenario where this is shown empty, for an account that HAS blogs.
+    ///
+    /// In order to adhere to this logic, if this VC is shown without a blog being set, we will try to load the "main" blog (ie in order: the last used blog,
+    /// the account's primary blog, or the first blog we find for the account).
+    ///
+    private func showBlogDetailsForMainBlogOrNoSites() {
+        guard let mainBlog = mainBlog() else {
+            showNoSites()
+            return
+        }
+
+        showBlogDetails(for: mainBlog)
+    }
+
+    // MARK: - No Sites UI logic
+
+    private func hideNoSites() {
+        hideNoResults()
+    }
+
+    private func showNoSites() {
+        hideBlogDetails()
+
+        addMeButtonToNavigationBar(email: defaultAccount()?.email, meScenePresenter: meScenePresenter)
+
+        configureAndDisplayNoResults(
+            on: view,
+            title: NSLocalizedString(
+                "Create a new site for your business, magazine, or personal blog; or connect an existing WordPress installation.",
+                comment: "Text shown when the account has no sites."),
+            buttonTitle: NSLocalizedString(
+                "Add new site",
+                comment: "Title of button to add a new site."),
+            image: "mysites-nosites")
+    }
+
+    // MARK: - Blog Details UI Logic
+
+    private func hideBlogDetails() {
+        guard let blogDetailsViewController = blogDetailsViewController else {
+            return
+        }
+
+        remove(blogDetailsViewController)
+        self.blogDetailsViewController = nil
+    }
+
+    /// Shows a `BlogDetailsViewController` for the specified `Blog`.  If the VC doesn't exist, this method also takes care
+    /// of creating it.
+    ///
+    /// - Parameters:
+    ///         - blog: The blog to show the details of.
+    ///
+    private func showBlogDetails(for blog: Blog) {
+        let blogDetailsViewController = self.blogDetailsViewController(for: blog)
+
+        addMeButtonToNavigationBar(email: blog.account?.email, meScenePresenter: meScenePresenter)
+
+        add(blogDetailsViewController)
+
+        blogDetailsViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        view.pinSubviewToAllEdges(blogDetailsViewController.view)
+    }
+
+    private func blogDetailsViewController(for blog: Blog) -> BlogDetailsViewController {
+        guard let blogDetailsViewController = blogDetailsViewController else {
+            let blogDetailsViewController = makeBlogDetailsViewController(for: blog)
+            self.blogDetailsViewController = blogDetailsViewController
+            return blogDetailsViewController
+        }
+
+        blogDetailsViewController.blog = blog
+        return blogDetailsViewController
+    }
+
+    private func makeBlogDetailsViewController(for blog: Blog) -> BlogDetailsViewController {
+        let blogDetailsViewController = BlogDetailsViewController(meScenePresenter: MeScenePresenter())
```

Comment:
we are not using all the benefits of dependency injection here, since we are directly instantiating one concrete type. We could use some sort of `factory` that this class uses to instantiate the presenter, if we want to abstract away this, but it might be overkilling.

Reply:
Once the feature flag is removed, `BlogDetailsViewController` will / should only be instantiated / shown by `MySiteViewController`.  It could even be made private to this class.

Due to this, I don't think there's a need to abstract its instantiation, as it should not be used elsewhere, and changing the VC internally for `MySiteViewController` would be a one-liner.

Reply:
Sounds good, I guess we can still use the `ScenePresenter` that is injected in `MySiteViewController`?

Reply:
Apologies, I misunderstood the feedback.  You're right.